service/imaging/processor_default.go

```diff
@@ -72,22 +74,50 @@ func (p *defaultProcessor) Fit(src io.ReadSeeker, width, height int) (image.Imag
 	return orig, nil
 }
 
-func (p *defaultProcessor) FitAnimationGIF(src io.Reader, width, height int) (*bytes.Reader, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second) // 10秒以内に終わらないファイルは無効
-	defer cancel()
-
-	b, err := imaging2.ResizeAnimationGIF(ctx, p.c.ImageMagickPath, src, width, height, false)
+func (p *defaultProcessor) FitAnimationGIF(srcFile io.Reader, width, height int) (*bytes.Reader, error) {
+	src, err := gif.DecodeAll(srcFile)
 	if err != nil {
-		switch err {
-		case context.DeadlineExceeded:
-			return nil, ErrTimeout
-		case imaging2.ErrInvalidImageSrc:
-			return nil, ErrInvalidImageSrc
-		default:
-			return nil, err
-		}
+		return nil, ErrInvalidImageSrc
+	}
+
+	srcWidth, srcHeight := src.Config.Width, src.Config.Height
+	// 画素数チェック
+	if srcWidth*srcHeight > p.c.MaxPixels {
+		return nil, ErrPixelLimitExceeded
 	}
-	return b, nil
+	// 画像が十分小さければスキップ
+	if srcWidth <= width && srcHeight <= height {
+		return imaging2.GifToBytesReader(src)
+	}
+
+	// 元の比率を保つよう調整
+	floatSrcWidth, floatSrcHeight, floatWidth, floatHeight := float64(srcWidth), float64(srcHeight), float64(width), float64(height)
+	if floatSrcWidth/floatSrcHeight > floatWidth/floatHeight {
+		height = int(math.Round(floatSrcHeight * floatWidth / floatSrcWidth))
+	} else if floatSrcWidth/floatSrcHeight < floatWidth/floatHeight {
+		width = int(math.Round(floatSrcWidth * floatHeight / floatSrcHeight))
+	}
+
+	dst := &gif.GIF{
+		Delay:     src.Delay,
+		LoopCount: src.LoopCount,
+		Disposal:  src.Disposal,
+		Config: image.Config{
+			ColorModel: src.Config.ColorModel,
+			Width:      width,
+			Height:     height,
+		},
+		BackgroundIndex: src.BackgroundIndex,
+	}
+
+	rect := image.Rect(0, 0, width, height)
+	for _, s := range src.Image {
+		d := image.NewPaletted(rect, s.Palette)
+		mks2013FilterKernel.Scale(d, rect, s, s.Bounds(), draw.Src, nil)
+		dst.Image = append(dst.Image, d)
+	}
```

Comment:
変数名はわかりやすく
フレームを表すならそう表してほしい(間違ってたらごめん)
```suggestion
	targetBounds := image.Rect(0, 0, width, height)
	for _, frame := range src.Image {
		destFrame := image.NewPaletted(rect, frame.Palette)
		mks2013FilterKernel.Scale(destFrame, targetBounds, frame, frame.Bounds(), draw.Src, nil)
		dst.Image = append(dst.Image, destFrame)
	}
```

Reply:
それで完璧です、ありがとうございます